BinaryObjectScanner.Protection/Macrovision.RipGuard.cs

```diff
@@ -25,6 +26,16 @@ internal string RipGuardCheckPortableExecutable(string file, PortableExecutable
             if (sections == null)
                 return null;
 
+            FileInfo fi = new FileInfo(file);
+
+            // So far, every seemingly-randomly named EXE on RipGuard discs have a consistent hash.
+            if (fi.Length == 49_152)
+            {
+                string sha1 = GetFileSHA1(file);
+                if (sha1 == "6A7B8545800E0AB252773A8CD0A2185CA2497938")
+                    return "RipGuard";
+            }
+
```

Comment:
Easy to forget, but for executable checks, the file is not guaranteed to exist. The way that this is being used right now means that it will throw an exception if you try to do the `new FileInfo` on a file that isn't there.

All other `GetFileSHA1` checks right now are in the path checks. If you choose to go this route, please wrap this entire thing in a `try/catch` where the `catch` is a no-op.